File: health_advanced_swift/HealthDashboard/Views/HDExerciseTypeView.m
//
//  HDExerciseTypeView.m
//  HealthDashboard
//
//  Created by zhang, haizi on 2026/3/21.
//

#import "HDExerciseTypeView.h"

@implementation HDExerciseTypeView

- (instancetype)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];
    if (self) {
        [self setupUI];
    }
    return self;
}

- (instancetype)initWithCoder:(NSCoder *)coder {
    self = [super initWithCoder:coder];
    if (self) {
        [self setupUI];
    }
    return self;
}

- (void)setupUI {
    self.layer.cornerRadius = 12;
    self.layer.masksToBounds = YES;
    self.backgroundColor = UIColor.secondarySystemBackgroundColor;
    
    // 标题
    _titleLabel = [UILabel new];
    _titleLabel.font = [UIFont systemFontOfSize:18 weight:UIFontWeightSemibold];
    _titleLabel.textColor = UIColor.labelColor;
    [self addSubview:_titleLabel];
    
    // 描述
    _descriptionLabel = [UILabel new];
    _descriptionLabel.font = [UIFont systemFontOfSize:14 weight:UIFontWeightRegular];
    _descriptionLabel.textColor = UIColor.secondaryLabelColor;
    _descriptionLabel.numberOfLines = 2;
    [self addSubview:_descriptionLabel];
    
    // 开始按钮
    _startButton = [UIButton buttonWithType:UIButtonTypeSystem];
    _startButton.titleLabel.font = [UIFont systemFontOfSize:16 weight:UIFontWeightSemibold];
    [_startButton setTitle:@"开始" forState:UIControlStateNormal];
    _startButton.layer.cornerRadius = 8;
    _startButton.backgroundColor = UIColor.systemBlueColor;
    [_startButton setTitleColor:UIColor.whiteColor forState:UIControlStateNormal];
    [self addSubview:_startButton];
}

- (void)layoutSubviews {
    [super layoutSubviews];
    
    CGFloat padding = 16;
    CGFloat width = self.bounds.size.width - padding * 2;
    
    _titleLabel.frame = CGRectMake(padding, padding, width, 24);
    _descriptionLabel.frame = CGRectMake(padding, padding + 28, width, 40);
    _startButton.frame = CGRectMake(padding, self.bounds.size.height - padding - 44, width, 44);
}

- (void)applyTheme {
    self.backgroundColor = UIColor.secondarySystemBackgroundColor;
    _titleLabel.textColor = UIColor.labelColor;
    _descriptionLabel.textColor = UIColor.secondaryLabelColor;
    _startButton.backgroundColor = UIColor.systemBlueColor;
}

@end

File: health_advanced_swift/HealthDashboard/Views/HDExerciseTypeView.swift
//
//  HDExerciseTypeView.swift
//  HealthDashboard - 运动类型选择卡片
//
//  Created by AI Assistant on 2026/3/22.
//

import UIKit

/// 运动类型选择卡片
class HDExerciseTypeView: UIView {

    // MARK: - Properties

    /// 运动类型（0=目标跑, 1=自由跑）
    var exerciseType: Int = 0

    // MARK: - UI

    let titleLabel = UILabel()
    let descriptionLabel = UILabel()
    let startButton = UIButton(type: .custom)

    // MARK: - Init

    override init(frame: CGRect) {
        super.init(frame: frame)
        setupUI()
    }

    required init?(coder: NSCoder) {
        super.init(coder: coder)
        setupUI()
    }

    // MARK: - UI Setup

    private func setupUI() {
        layer.cornerRadius = 12
        layer.masksToBounds = true
        backgroundColor = .secondarySystemBackground

        titleLabel.font = .systemFont(ofSize: 18, weight: .semibold)
        titleLabel.textColor = .label
        addSubview(titleLabel)

        descriptionLabel.font = .systemFont(ofSize: 14)
        descriptionLabel.textColor = .secondaryLabel
        descriptionLabel.numberOfLines = 2
        addSubview(descriptionLabel)

        startButton.setTitle("开始", for: .normal)
        startButton.titleLabel?.font = .systemFont(ofSize: 16, weight: .semibold)
        startButton.layer.cornerRadius = 8
        startButton.backgroundColor = .systemBlue
        startButton.setTitleColor(.white, for: .normal)
        addSubview(startButton)
    }

    override func layoutSubviews() {
        super.layoutSubviews()
        let padding: CGFloat = 16
        let width = bounds.width - padding * 2
        titleLabel.frame = CGRect(x: padding, y: padding, width: width, height: 24)
        descriptionLabel.frame = CGRect(x: padding, y: padding + 28, width: width, height: 40)
        startButton.frame = CGRect(x: padding, y: bounds.height - padding - 44, width: width, height: 44)
    }

    // MARK: - Theme

    @objc func applyTheme() {
        backgroundColor = .secondarySystemBackground
        titleLabel.textColor = .label
        descriptionLabel.textColor = .secondaryLabel
        startButton.backgroundColor = .systemBlue
    }
}
